### PR TITLE
Add clone for repatched components

### PR DESCRIPTION
**Pull request by mseri**

### CODE REVIEW

File: planex/cmd/fetch.py [hunk 1]
```diff
@@ -213,20 +237,11 @@ def fetch_source(args):
...
-        shutil.copyfile(url.path, resource.path)
-        write_originfile(resource.path, url.path)
-
-    else:
+    try:
+        fetch_source_dispatch(resource, args.retries)
+    except UnsupportedScheme as exn:
         sys.exit("%s: Unsupported url scheme %s" %
-                 (sys.argv[0], url.scheme))
+                 (sys.argv[0], exn.message))
```
Comment (srowe) [inline planex/cmd/fetch.py]: `exception.message` is deprecated

https://www.python.org/dev/peps/pep-0352/#retracted-ideas

File: planex/cmd/fetch.py [hunk 2]
```diff
@@ -195,6 +195,30 @@ def fetch_repo(url, resource):
...
 
 
+class UnsupportedScheme(Exception):
+    """UnsupportdScheme exception"""
+    pass
+
+
+def fetch_source_dispatch(resource, retries):
+    """
+    Dispatch to the appropriate fetch metod for the provided resource.
```
Comment (srowe) [inline planex/cmd/fetch.py]: "method"

**Comment by mseri** [comment]
This is now assemblying correctly patchqueues and repatched components for all the tested components

File: planex/cmd/fetch.py [hunk 3]
```diff
@@ -194,6 +194,30 @@ def fetch_repo(url, resource):
     write_originfile(resource.path, repo.remotes.origin.url, sha)
 
 
+class UnsupportedScheme(Exception):
+    """UnsupportdScheme exception"""
```
Comment (srowe) [inline planex/cmd/fetch.py]: "UnsupportedScheme"

Comment (mseri) [reply planex/cmd/fetch.py]: Thanks, I have pushed the fix typo to depend-defactor

**Pull request merged**